internal/gatewayapi/envoypatchpolicy.go

```diff
@@ -39,7 +46,54 @@ func (t *Translator) ProcessEnvoyPatchPolicies(envoyPatchPolicies []*egv1a1.Envo
 		irKey := irStringKey(string(*targetNs), string(policy.Spec.TargetRef.Name))
 		gwXdsIR, ok := xdsIR[irKey]
 		if !ok {
-			// TODO: Update Status
+			// This status condition will not get updated in the resource because
+			// the IR is missing, but it has been kept here in case we publish
+			// the status from this layer instead of the xds layer.
+			message := fmt.Sprintf("Gateway:%s not found.", policy.Spec.TargetRef.Name)
+
+			status.SetEnvoyPatchPolicyCondition(policy,
+				gwv1a2.PolicyConditionAccepted,
+				metav1.ConditionFalse,
+				gwv1a2.PolicyReasonTargetNotFound,
+				message,
+			)
+			continue
+		}
+
+		// Create the IR with the context need to publish the status later
+		policyIR := ir.EnvoyPatchPolicy{}
+		policyIR.Name = policy.Name
+		policyIR.Namespace = policy.Namespace
+		policyIR.Status = &policy.Status
+
+		// Append the IR
+		gwXdsIR.EnvoyPatchPolicies = append(gwXdsIR.EnvoyPatchPolicies, &policyIR)
+
+		// Ensure policy can only target a Gateway
+		if policy.Spec.TargetRef.Group != gwv1b1.GroupName || policy.Spec.TargetRef.Kind != KindGateway {
+			message := fmt.Sprintf("TargetRef.Group:%s TargetRef.Kind:%s, only TargetRef.Group:%s and TargetRef.Kind:%s is supported.",
+				policy.Spec.TargetRef.Group, policy.Spec.TargetRef.Kind, gwv1b1.GroupName, KindGateway)
+
+			status.SetEnvoyPatchPolicyCondition(policy,
+				gwv1a2.PolicyConditionAccepted,
+				metav1.ConditionFalse,
+				gwv1a2.PolicyReasonInvalid,
+				message,
+			)
+			continue
+		}
+
+		// Ensure Policy and target Gateway are in the same namespace
+		if policy.Namespace != string(*targetNs) {
```

Comment:
Same question as the above comment for this one

Reply:
same as https://github.com/envoyproxy/gateway/pull/1691#discussion_r1274218967